### PR TITLE
Release/v3

### PR DESCRIPTION
**Pull request by DarkoKukovec**
Please select all that apply:

* [ ] This PR contains a new feature
* [x] This PR updates an existing feature
* [ ] This PR contains bugfixes
* [x] This PR contains all the relevant tests
* [x] This PR creates a breaking change

Please describe the differences between the current and new behavior:

Removes dependency from mobx, basically same as v2 with disable-mobx, but without extra config steps.

Once ready, it should be merged, but the branch should not be deleted.

TODO:
* [x] Fix the build step (flaky)
* [x] Verify all tests work
* [x] Fix checks
* [x] Fix SonarCloud issues

### CODE REVIEW

**Comment by DarkoKukovec** [comment]
@fvoska @isBatak there are a lot of changes here and I don't think it makes sense to go through all of them. This is more of a sanity check, if you see any issues (you can focus more on angular and swr).

SonarCloud detected a bunch of code smells, but they're not new - they have been surfaced when I fixed linter issues. Since I don't see anything critical there, I would ignore them for now as not to block the release.

File: packages/datx-jsonapi-angular/package.json [hunk 1]
```diff
@@ -1,8 +1,32 @@
 {
   "name": "@datx/jsonapi-angular",
-  "version": "2.6.1",
+  "version": "3.0.0-beta.0",
+  "description": "DatX mixin for Angular JSON API support",
+  "main": "dist/bundles/datx-jsonapi-angular.umd.js",
```
Comment (fvoska) [inline packages/datx-jsonapi-angular/package.json:5]: This should not be here. This is what the build process generates and should be present only in the generated dist/package.json file. Not sure how it got commited to this package.json file.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -10,31 +10,16 @@ jobs:
...
-      - name: Cleanup MobX
-        if: ${{ matrix.mobx-version == '0' }}
-        run: yarn workspace @datx/utils remove mobx
-      - name: Install correct MobX version
-        if: ${{ matrix.mobx-version > 0 }}
-        run: yarn workspace @datx/utils add mobx@${{ matrix.mobx-version }} --dev
-      - run: lerna run build --no-private
-      - run: MOBX_VERSION=${{ matrix.mobx-version }} lerna run test --no-private
+      - run: lerna run build
+      - run: lerna run test
```
Comment (fvoska) [inline .github/workflows/tests.yml:25]: You dropped lerna from husky pre-commit hook. Why not use npm run here as well?

File: lerna.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "lerna": "2.5.1",
```
Comment (fvoska) [inline lerna.json:2]: Installed lerna version is much newer. Not use if it matters to update the version here as well.

It's not strictly related to the changeset of this PR and it was like that before as well, but I noticed it now. Might be a good time to check it

**Review by fvoska — CHANGES_REQUESTED** [review]
Reviewed Angular part

File: packages/datx-jsonapi-angular/package.json (hunk 1 above)
Comment (DarkoKukovec) [reply packages/datx-jsonapi-angular/package.json:5]: I probably added it manually during conflict resolution. Will remove it again

**Review by isBatak — APPROVED** [review]

**Review by fvoska — APPROVED** [review]

**Pull request merged**